File: composeApp/src/desktopMain/kotlin/fr/thomasbernard03/androidtools/domain/usecases/file/DownloadFileUseCase.kt
package fr.thomasbernard03.androidtools.domain.usecases.file

import fr.thomasbernard03.androidtools.domain.repositories.FileRepository
import org.koin.java.KoinJavaComponent.get

class DownloadFileUseCase(
    private val fileRepository: FileRepository = get(FileRepository::class.java)
) {
    suspend operator fun invoke(path: String, targetPath: String) {
        fileRepository.downloadFile(path, targetPath)
    }
}

File: lib/features/file_explorer/domain/usecases/download_file_usecase.dart
import 'package:android_tools/features/file_explorer/domain/repositories/file_repository.dart';

class DownloadFileUsecase {
  final FileRepository _fileRepository;

  DownloadFileUsecase(this._fileRepository);

  Future<void> call(String filePath, String destinationPath, String deviceId) {
    return _fileRepository.downloadFile(filePath, destinationPath, deviceId);
  }
}
